src/components/table/table.scss

```diff
@@ -0,0 +1,86 @@
+@use "../../design-tokens/colors.yml";
+@use "../../design-tokens/font.yml";
+@use "../../design-tokens/sizes.yml";
+@use "../../mixins/ms";
+
+$table-border-color: colors.$gray-light;
+$table-border-thin: sizes.$edge_small solid $table-border-color;
+$table-border-thick: sizes.$edge_medium solid $table-border-color;
+
+/**
+ * Our base table component
+ *
+ * Since tables require specific markup using the BEM element class names can
+ * quickly become overwhelming. We make an exception from our BEM rules to
+ * directly style elements within `.c-table`s.
+ */
+
+/**
+ * 1. Tables may overflow horizontally. Keep this from breaking layouts while
+ *    ensuring content is viewable
+ *    In the future we may want to consider making it visually obvious when
+ *    tables are clipped.
+ */
+
+.c-table {
+  overflow-x: auto; /* 1 */
+}
+
+/**
+ * Table cells and headers share padding
+ */
+.c-table td,
+.c-table th {
+  padding: ms.step(-3);
+}
+
+/**
+ * Table headings are semibold to stand out as labels
+ */
+.c-table th {
+  font-weight: font.$weight-semibold;
+}
+
+/**
+ * Table headers and footers are separated from the table body with strokes.
+ */
+.c-table thead {
+  border-bottom: $table-border-thick;
+}
+
+.c-table tfoot {
+  border-top: $table-border-thick;
+}
+
+/**
+ * Table rows can be styled in a couple ways.
+ * This is achieved by adding a class to the table itself.
+ */
+.c-table--ruled tbody tr {
+  border-bottom: $table-border-thin;
+}
+
+.c-table--striped tbody tr:nth-of-type(2n - 1) {
+  background-color: colors.$gray-lighter;
+}
+
+/**
+ * When comparing numeric table cells, it's easier if the ones, tens, hundreds,
+ * etc. columns are lines up.
+ *
+ * 1. Make numbers equal width so the columns line up
+ * 2. Right align numeric cells so the "ones" column lines up on the right side
+ */
+.c-table__numeric-cell {
+  font-variant-numeric: tabular-nums; /* 1 */
+  text-align: right; /* 2 */
+}
+
+/**
+ * Optionally right align row headings.
+ * This keeps headings closer to the content they label and in some cases
+ * improves scannability.
+ */
+.c-table--tighten-row-headings [scope='row'] {
+  text-align: right;
+}
```

Comment:
This might be crazy, but do you think there's a way to apply a modifier to `c-table` that applies this change to `th[scope='row']` elements and the `numeric-cell` modifier to `td` elements?

Reply:
I wondered about this as well. My concern was that your table may have some columns that are numeric data and others that are strings (so shouldn't be right aligned or have tabular numbers). I guess it might be good to have the option to either apply this at the cell level or on the table level?

Reply:
I'm just trying to think of adding these via a CMS. If the feature is available as a top-level modifier, the WordPress Gutenberg table block can support it (every Gutenberg block has a "class" field). But if it's only available on cells, then it will require editing the block as HTML, writing a custom Gutenberg block or manually coding the table.

Reply:
Yeah that's a good point. I hadn't been considering it from that perspective. I'd been imagining populating via JSON or hand-building for blog posts but you're right that it would be nice to be able to use this with the Gutenberg block building UI. I'll look into combining these and applying the class at the table level.